tests/cli-self-upd.rs

```diff
@@ -328,12 +328,24 @@ fn install_adds_path_to_profile() {
     install_adds_path_to_rc(".profile");
 }
 
+#[test]
+#[cfg(unix)]
+fn install_adds_path_to_bashrc() {
+    install_adds_path_to_rc(".bashrc");
+}
+
 #[test]
 #[cfg(unix)]
 fn install_adds_path_to_bash_profile() {
     install_adds_path_to_rc(".bash_profile");
 }
 
+#[test]
+#[cfg(unix)]
+fn install_adds_path_to_zshenv() {
```

Comment:
I'd really like to see these written as unit tests: right now they will run serially due to the mutex in setup(); they are doing about 10k times the work actually under test.

Alternatively, just have one test that has all the files on disk and check they are all modified.

You can see a few recent patches from me, including the unmerged https://github.com/rust-lang/rustup/pull/2412 for some possible sorts of changes to enable this: but basically separating out functional and pure layers and testing logic via pure functions (where possible).

Reply:
Cutting down the work sounds good. Pulling them into a single test seems like the current path of least resistance given how much of it is testing on doing IO (therefore impure) in the correct places in the correct manner?

Reply:
I'm sorry, I have tried several times to understand it but I simply do not get the unit test framework well enough at all to write these as unit tests. I will not be doing so for this patch.